sbnobj/Common/Calibration/TrackCaloSkimmerObj.h

```diff
@@ -0,0 +1,156 @@
+#ifndef SBN_TrackCaloSkimmerObj
+#define SBN_TrackCaloSkimmerObj
+
+#include <vector>
+
+namespace sbn {
+  struct HitInfo {
+    float integral;
+    float sumadc;
+    float width;
+    float pitch;
+    float dqdx;
+    float rr;
+    float x;
+    float y;
+    float z;
+    float time;
+    uint16_t wire;
+    uint16_t plane;
+    uint16_t tpc;
+    uint16_t mult;
+    uint16_t i_snippet;
+    int16_t start;
+    int16_t end;
+    bool ontraj;
+    bool oncalo;
+
+    // raw adc info
+    short tdc0;
+    std::vector<short> adcs;
+
+    HitInfo():
+      integral(-1),
+      sumadc(-1),
+      width(-1),
+      pitch(-1),
+      dqdx(-1),
+      rr(-1),
+      x(-1),
+      y(-1),
+      z(-1),
+      time(-1),
+      wire((uint16_t)-1),
+      plane((uint16_t)-1),
+      tpc((uint16_t)-1),
+      mult((uint16_t)-1),
+      i_snippet((uint16_t)-1),
+      start(-1),
+      end(-1),
+      ontraj(false),
+      oncalo(false),
+      tdc0(-1) {}
+  };
+
+  struct MetaInfo {
+    int run;
+    int evt;
+    int subrun;
+    int ifile;
+    int iproc;
+
+   MetaInfo():
+     run(-1),
+     evt(-1),
+     subrun(-1),
+     ifile(-1),
+     iproc(-1) {}
+  };
+
+  struct TrackInfo {
+    MetaInfo meta;
+    std::vector<HitInfo> hits0; 
+    std::vector<HitInfo> hits1; 
+    std::vector<HitInfo> hits2; 
+    float t0; 
+    int id;
+    bool clear_cosmic_muon;
+    float start_x; 
+    float start_y;
+    float start_z;
+    float end_x; 
+    float end_y;
+    float end_z;
+    float dir_x; 
+    float dir_y;
+    float dir_z;
+    float length;
+
+    unsigned ndaughters;
+
+    float hit_min_time_p0_tpcE;
```

Comment:
Actually, a little confused here. this is per plane, but then also ... per cryostat? there are two physical TPCs in east and west, cryostats, so not sure what these hits correspond to?

Might suggest that this be keyed by PlaneID (e.g. a map of floats?) to make it a bit more general to SBND too. Think ROOT should handle maps ok.

Reply:
It's per-plane per-TPC (E/W). I believe that structure should also generalize to SBND.

Reply:
Sorry, maybe I'm misunderstanding: is this already per cryostat somehow? So, one would do this for the west cryostat, and then again for the easy cryostat, and those would be two separate collections?

Reply:
Each TrackInfo object corresponds to a single reconstructed track, and each track is always in one cryostat. So it is "per-cryostat" in that sense.

Reply:
OK, I still suggest this be keyed per PlaneID, as that will be general and is what you want. It may be very much useful to know which cryostat the tracks you're looking at come from (e.g. purity is different), and so I would not lose that information.

Reply:
I want to be able to access this data outside of the LArSoft context, which is why I have avoided using any LArSoft-specific objects. But I'll add in a cryostat field.

Reply:
Fixed.

Reply:
OK, One last thing is that I noticed the cryostat is not included in the HitInfo, which I think you want to add.

Reply:
hmm... I don't think that's necessary.

Right now the HitInfo is the most common object, so I want to keep its size as small as possible. And tracks are by definition limited to a cryostat, so the hits are all in the same cryostat as the track. So one can obtain the hit Cryostat from the TracjInfo.

Reply:
Fair enough.